transport/http/src/main/java/org/kaazing/gateway/transport/http/bridge/filter/HttpLoginSecurityFilter.java

```diff
@@ -288,24 +288,34 @@ protected boolean login(NextFilter nextFilter,
                      rolesAreSufficient);
         }
 
+        String clientChallengeScheme = authToken.getScheme();
+        String expectedChallengeScheme = getBaseAuthScheme(realm.getChallengeScheme());
+
         // We have to issue a challenge when ALL of the below hold:
         //
         // - no authorization token was presented
         // - not all required roles (if any) are present
         //
         // Issue a challenge since authentication is required.
-        if (authTokenIsMissing(authToken) &&
-            rolesAreSufficient == false) {
+        if ((authTokenIsMissing(authToken) &&
+            rolesAreSufficient == false) ) {
             return loginMissingToken(nextFilter, session, httpRequest, authToken, additionalCallbacks, realms, realmIndex, loginContexts);
         }
 
         DefaultLoginResult loginResult = null;
+        ResultAwareLoginContext loginContext = null;
+
+        if (clientChallengeScheme != null && clientChallengeScheme.equals(expectedChallengeScheme) == false) {
```

Comment:
Note, I believe [RFC 7235](https://tools.ietf.org/html/rfc7235#section-2.1) states scheme is case-insensitive 

```
It uses a case-
   insensitive token as a means to identify the authentication scheme,
   followed by additional information necessary for achieving




Fielding & Reschke           Standards Track                    [Page 3]
 
RFC 7235                 HTTP/1.1 Authentication               June 2014


   authentication via that scheme.  The latter can be either a comma-
   separated list of parameters or a single sequence of characters
   capable of holding base64-encoded information.
```

so maybe equalsIgnoreCase?

Please add a k3po spec test for this.